pkg/apiclients/testapi/testapi.go

```diff
@@ -0,0 +1,472 @@
+	resp, err := h.client.lowLevelClient.GetTestWithResponse(ctx, h.orgID, testID, getTestParams)
+	if err != nil {
+		return nil, fmt.Errorf("get test %s request failed: %w", testID, err)
+	}
+
+	if resp.ApplicationvndApiJSON200 == nil {
+		return nil, fmt.Errorf("test %s had unexpected status %d or wrong response body format", testID, resp.StatusCode())
+	}
+
+	testData := resp.ApplicationvndApiJSON200.Data
+	attrs := testData.Attributes
+	status := &Result{
+		TestID:           testData.Id,
+		EffectiveSummary: attrs.EffectiveSummary,
+		RawSummary:       attrs.RawSummary,
+		handle:           h, // Populate the handle field
+	}
+
+	if attrs.State != nil {
+		status.State = string(attrs.State.Execution)
+		if attrs.State.Errors != nil && len(*attrs.State.Errors) > 0 {
+			status.Errors = attrs.State.Errors
+		}
+		if attrs.State.Warnings != nil && len(*attrs.State.Warnings) > 0 {
+			status.Warnings = attrs.State.Warnings
+		}
+	} else {
+		status.State = "unknown" // API spec defines this as always set
+	}
+
+	if attrs.Outcome != nil {
+		status.Outcome = &attrs.Outcome.Result
+		status.OutcomeReason = attrs.Outcome.Reason
+	}
+
+	return status, nil
+}
+
+// Returns a channel signaling completion of Wait()
+func (h *testHandle) Done() <-chan struct{} { return h.doneChan }
+
+func (h *testHandle) Result() *Result {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	return h.result
+}
+
+func (h *testHandle) setResult(status *Result) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	h.result = status
+}
+
+// Findings fetches and returns the findings associated with this test result.
+// It fetches findings lazily on the first call. Subsequent calls return cached data.
+//   - resultFindings: The list of findings.
+//   - complete: True if all pages of findings were successfully fetched. False if an error
+//     occurred during pagination or if the context was canceled, in which case
+//     'resultFindings' may be partial.
+//   - err: Any error encountered during the fetching process.
+func (r *Result) Findings(ctx context.Context) (resultFindings []FindingData, complete bool, err error) {
+	if r.TestID == nil {
+		return nil, false, fmt.Errorf("TestID is nil, cannot fetch findings")
+	}
+	if r.handle == nil {
+		return nil, false, fmt.Errorf("internal error: handle not set in Result, cannot access client/orgID for findings")
+	}
+
+	r.findingsOnce.Do(func() {
+		fetchedData, isComplete, fetchErr := r.fetchFindingsInternal(ctx)
+		r.findings = fetchedData
+		r.findingsComplete = isComplete
+		r.findingsError = fetchErr
+	})
+	return r.findings, r.findingsComplete, r.findingsError
+}
+
+// fetchFindingsInternal is an unexported method on Result that contains the actual
+// pagination and fetching logic.
+func (r *Result) fetchFindingsInternal(ctx context.Context) ([]FindingData, bool, error) {
+	if r.TestID == nil {
+		return nil, false, fmt.Errorf("cannot fetch findings, TestID is nil in internal method")
```

Comment:
Suggestion: There seem to be some duplications in errors with slight variation. Please try to unify them to reduce cardinality.